File: solutions/javascript/pig-latin/1/pig-latin.js
// A needlessly convoluted and unreadable pig-latin solution :)

class PigLatin {
	constructor(regex, vowel = false) {
		this.rule = new RegExp(`^([${vowel ? "" : "^"}aeiou]${regex})(.+)?`);
	}

	test(str) {
		return this.rule.test(str);
	}

	replace(str, pattern) {
		return str.replace(this.rule, pattern);
	}
}

const rule1 = new PigLatin("|xr|yt", true),
	rule2 = new PigLatin("+"),
	rule3 = new PigLatin("{0,}qu"),
	rule4 = new PigLatin("+?)(y");

/**
 * Translates a string to Pig Latin
 * @param {string} english
 * @returns {string}
 */
export const translate = (english) => {
	return english
		.split(" ")
		.map((w) => {
			for (let i = 0; i < 4; i++) {
				let rule = [rule4, rule1, rule3, rule2][i];
				if (rule.test(w))
					return rule.replace(w, ["$2$3$1ay", "$1$2ay", "$2$1ay", "$2$1ay"][i]);
			}

			return w;
		})
		.join(" ");
};

File: solutions/python/pig-latin/1/pig_latin.py
def translate(text):
    textarr = text.split()
    vowels = ['a', 'e', 'i', 'o', 'u']
    res = []
    for i in textarr:
        if (i[0] in vowels or (i.startswith("xr") or i.startswith("yt"))):
            res.append(f"{i}ay")
            break
        if (i[1] == "q" and i[2] == "u"):
            res.append(f"{i[3:]}{i[:3]}ay")
            break

        for j in i:
            if j in vowels or j == "y":
                index = i.index(j)

                if index == 0:
                    index = 1

                if (i.startswith("qu") and j == "u"):
                    continue

                res.append(f"{i[(index):]}{i[:index]}ay")
                print(res)
                break

    return " ".join(res)
